ops/framework.py

```diff
@@ -1054,6 +1055,10 @@ def __eq__(self, other):
         else:
             return NotImplemented
 
+    def copy(self) -> typing.Dict:
+        """Return a copy of the backing data as a dict."""
+        return dict(self._under)
```

Comment:
Maybe the result of this should be also a `StoredDict`, not a simple `dict`?

Reply:
The original request from the issue was to return a `dict`. Maybe a better idea to rename this to `as_dict()` or similar, but then we lose semantic compatibility

Reply:
Maybe have both copy and as_dict?
Seems odd that type(o) != type(o.copy()) and probably also not isinstance nor issubclass.

Reply:
I'm hesitant to implement `.copy()` primarily because the semantics of a copied `dict` are somewhat different and unintuitive. For example, [this](https://github.com/canonical/operator/issues/587#issuecomment-903846086).

Added, but needs further discussion.

Reply:
After talking a little about this, +1 to return a dict, so resolving this conversation...